I'm not sure what's wrong here, but a stream unit test is now failing for some reason:

```
======================================================================
ERROR: test_modify_stream (tests.roundwared.test_stream.TestRoundStream)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vagrant/tests/roundwared/test_stream.py", line 54, in test_modify_stream
    stream.modify_stream(req)
  File "/vagrant/roundwared/stream.py", line 110, in modify_stream
    filenames = self.recordingCollection.get_filenames()
  File "/vagrant/roundwared/recording_collection.py", line 176, in get_filenames
    self.playlist_proximity)
TypeError: argument 2 to map() must support iteration

----------------------------------------------------------------------
```

Looks like my new ordering function somehow yields a `playlist_proximity` that is no longer iterable?
Oddly, I didn't get this failure with my original code duplication fix to this issue and I thought this new approach was essentially the same, but more efficient and sensible from a coding perspective.
Ideas?

--- a/roundwared/recording_collection.py
+++ b/roundwared/recording_collection.py
@@ -184,6 +184,19 @@ class RecordingCollection:
         """
         return (len(self.banned_proximity) > 0 and len(self.banned_timeout) > 0)
 
+    def order_assets(self, assets):
+        """
+        Order assets by ordering method set in project config
+        """
+        logger.debug("Ordering assets by: %s" % self.ordering)
+        if self.ordering == 'random':
+            return order_assets_randomly(assets)
+        elif self.ordering == 'by_like':
+            return order_assets_by_like(assets)
+        elif self.ordering == 'by_weight':
+            return order_assets_by_weight(assets)
+        return []
+
     def _update_playlist_proximity(self, request):
         current_time = time()
         # Remove all expired asset bans from the ban list.
@@ -213,13 +226,8 @@ class RecordingCollection:
             if not self._banned(r) and self._is_nearby(request, r):
                 self.playlist_proximity.append(r)
 
-        logger.debug('Ordering is: ' + self.ordering)
-        if self.ordering == 'random':
-            self.playlist_proximity = order_assets_randomly(self.playlist_proximity)
-        elif self.ordering == 'by_like':
-            self.playlist_proximity = order_assets_by_like(self.playlist_proximity)
-        elif self.ordering == 'by_weight':
-            self.playlist_proximity = order_assets_by_weight(self.playlist_proximity)
+        # apply project ordering
+        self.playlist_proximity = self.order_assets(self.playlist_proximity)
 
 
     def _is_nearby(self, request, recording):
@@ -260,5 +268,8 @@ class RecordingCollection:
         # and are in self.all, ensuring that tag filters are applied
         self.playlist_timed = [item.asset for item in playlist
                                if not self._banned(item.asset) and item.asset in self.all]
+
+        # apply project ordering
+        self.playlist_timed = self.order_assets(self.playlist_timed)
 
         logger.debug("Found timed assets: %s" % self.playlist_timed)

--- a/tests/roundwared/test_recording_collection.py
+++ b/tests/roundwared/test_recording_collection.py
@@ -86,6 +86,11 @@ class TestRecordingCollection(RoundwaredTestCase):
                                  tags=[self.tag1],
                                  audiolength=2000, weight=300,
                                  latitude=0.1, longitude=0.1)
+        self.asset7 = mommy.make(Asset, project=self.project3,
+                                 language=self.english,
+                                 tags=[self.tag2],
+                                 audiolength=2000, weight=100,
+                                 latitude=2.0, longitude=2.0)
         self.masterui1 = mommy.make(MasterUI, project=self.project1,
                                     ui_mode=MasterUI.LISTEN,
                                     tag_category=self.tagcat1)
@@ -102,9 +107,11 @@ class TestRecordingCollection(RoundwaredTestCase):
         self.uimapping3 = mommy.make(UIMapping, master_ui=self.masterui3,
                                      tag=self.tag2, default=True, active=True)
         self.timedasset1 = mommy.make(TimedAsset, project=self.project3,
-                                     asset=self.asset4, start=0, end=15)
+                                      asset=self.asset4, start=0, end=15)
         self.timedasset2 = mommy.make(TimedAsset, project=self.project3,
-                                     asset=self.asset6, start=0, end=30)
+                                      asset=self.asset6, start=0, end=30)
+        self.timedasset3 = mommy.make(TimedAsset, project=self.project3,
+                                      asset=self.asset7, start=0, end=30)
 
     def test_instantiate_recording_collection(self):
         req = self.req1
@@ -312,13 +319,13 @@ class TestRecordingCollection(RoundwaredTestCase):
         """ Test that _get_recording returns a proximity asset instead of a
             timed asset when project.timed_asset_priority=True.
         """
-
         stream = RoundStream(self.session3.id, 'ogg', self.req3)
         rc = RecordingCollection(stream, self.req3, stream.radius, 'by_weight')
         # Force/Fake start the RC timer; needed for timed asset selection
         rc.start()
         # Update the list of nearby recordings
         rc.update_request(self.req3)
+        self.assertEquals(self.asset7, rc.get_recording())
         self.assertEquals(self.asset4, rc.get_recording())
         self.assertEquals(self.asset5, rc.get_recording())
 
@@ -336,6 +343,7 @@ class TestRecordingCollection(RoundwaredTestCase):
         # Update the list of nearby recordings
         rc.update_request(self.req3)
         self.assertEquals(self.asset5, rc.get_recording())
+        self.assertEquals(self.asset7, rc.get_recording())
         self.assertEquals(self.asset4, rc.get_recording())
 
     def test_timed_assets_filtered_by_tags(self):
@@ -355,3 +363,19 @@ class TestRecordingCollection(RoundwaredTestCase):
         self.assertEquals(self.asset6, rc.get_recording())
         self.assertEquals(None, rc.get_recording())
 
+    def test_timed_asset_ordering_by_weight(self):
+        """ Test that timed assets are ordered per the project.ordering
+            parameter (in this case 'by_weight').
+        """
+        stream = RoundStream(self.session3.id, 'ogg', self.req3)
+        rc = RecordingCollection(stream, self.req3, stream.radius, 'by_weight')
+        # Force/Fake start the RC timer; needed for timed asset selection
+        rc.start()
+        # move to location with no geo-assets
+        self.req3['latitude'] = 10
+        # Update the list of nearby recordings
+        rc.update_request(self.req3)
+        # verify that asset with largest 'weight' value is returned first
+        self.assertEquals(self.asset7, rc.get_recording())
+        self.assertEquals(self.asset4, rc.get_recording())
+        self.assertEquals(None, rc.get_recording())

--- a/tests/roundwared/test_stream.py
+++ b/tests/roundwared/test_stream.py
@@ -16,7 +16,7 @@ class TestRoundStream(RoundwaredTestCase):
         super(type(self), TestRoundStream).setUp(self)
 
         self.project1 = mommy.make(Project, name='Project One',
-                                   ordering=1, recording_radius=16)
+                                   ordering='random', recording_radius=16)
         self.session1 = mommy.make(Session, project=self.project1,
                                    language=self.english)
         self.req1 = {"session_id": self.session1.id,
@@ -57,7 +57,7 @@ class TestRoundStream(RoundwaredTestCase):
         req = self.req1
         req["audio_stream_bitrate"] = '128'
         stream = RoundStream(self.session1.id, 'ogg', req)
-        self.assertEquals('1', stream.ordering)
+        self.assertEquals('random', stream.ordering)
 
     def test_main_loop(self):
         """ make sure we get a gObject MainLoop """

